### PR TITLE
feat: Add automatic price reduction on reposts

### PR DESCRIPTION
**Pull request by 1cu**
## ℹ️ Description

Implements automatic price reduction functionality that allows ads to gradually decrease in price on each repost until a minimum price is reached.

- Link to the related issue(s): #672 
- Motivation: Automate the common workflow of gradually reducing prices on reposts to attract buyers while maintaining pricing control

## 📋 Changes Summary

- Add automatic price reduction with PERCENTAGE and FIXED strategies
- Add configurable minimum price floor and delay mechanisms (repost count, days)
- **Refactored configuration structure** into nested `auto_price_reduction` object for better organization
- Add `AutoPriceReductionConfig` model with validation
- Add price reduction logic with delay/cycle tracking
- Fix metadata persistence timing (price_reduction_count only persisted after successful publish)
- Add warning for min_price == price misconfiguration
- Add 569 comprehensive unit tests covering all edge cases
- Update README.md with configuration examples (both PERCENTAGE and FIXED strategies)
- Add German translations for all user-facing messages
- Update JSON schemas for editor validation
- Code quality improvements: extract shared validation logic, add constants, reduce log noise

**Configuration format:**
```yaml
auto_price_reduction:
  enabled: true
  strategy: PERCENTAGE  # or FIXED
  amount: 10
  min_price: 90
  delay_reposts: 0
  delay_days: 0
```

**Dependencies**: None - uses existing `Decimal` for financial precision

### ⚙️ Type of Change
- [x] ✨ New feature (adds new functionality without breaking existing usage)
- [x] ♻️ Code refactoring (improved structure based on review feedback)
- [x] 🐞 Bug fix (metadata persistence timing fix)

## ✅ Checklist
- [x] I have reviewed my changes to ensure they meet the project's standards.
- [x] I have tested my changes and ensured that all tests pass (`pdm run test`).
- [x] I have formatted the code (`pdm run format`).
- [x] I have verified that linting passes (`pdm run lint`).
- [x] I have updated documentation where necessary.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

## Summary by CodeRabbit

* **New Features**
  * Automatic price reduction on reposts with configurable strategy (PERCENTAGE or FIXED), amount, min-price floor, and delays (by reposts or days); utility added to compute reduced prices.

* **Behavior**
  * Reductions run only on repost (replace) flows; repost and reduction counters tracked and advanced and persisted after successful publish.

* **Documentation**
  * README expanded with examples, rounding rules and delay guidance.

* **Tests**
  * Extensive unit tests for validation, reduction logic, delays, persistence, logging, and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -377,6 +377,11 @@ category: # e.g. "Elektronik > Notebooks"
 
 price: # without decimals, e.g. 75
 price_type: # one of: FIXED, NEGOTIABLE, GIVE_AWAY (default: NEGOTIABLE)
+auto_reduce_price: # true or false to enable automatic price reduction on reposts (default: false)
```
Comment (sebthom) [inline README.md]: wouldn't it make sense to group everything related to price reduction under one config key, e.g.

```yaml
auto_price_reduction:
  enabled: false
  strategy:
    type: PERCENTAGE
    amount: 10
  min_price: 90
  delay_reposts: 2
  delay_days: 0
```

or


```yaml
auto_price_reduction:
  enabled: false
  strategy: PERCENTAGE
  amount: 10
  min_price: 90
  delay_reposts: 2
  delay_days: 0
```

Comment (1cu) [reply README.md]: Your structure is better for these reasons:

  1. Better Grouping: All auto-pricing fields are nested under one key (auto_price_reduction), making it clearer which fields are related
  2. Cleaner Namespace: Doesn't pollute the top-level ad config with multiple price_reduction_* fields
  3. More Explicit: enabled: false is clearer than auto_reduce_price: true
  4. Better Naming: amount instead of value is more descriptive
  5. Easier to Extend: Future auto-pricing features would naturally fit under auto_price_reduction
  6. Less Redundant: The current structure has redundant "price_reduction" in multiple field names

Between the two options, I'd recommend Option 2 (flat strategy) because:
  - Simpler and easier to read/write
  - strategy: PERCENTAGE is clear enough without nesting
  - Fewer indentation levels
  - The nested strategy object in Option 1 only has 2 fields, which doesn't justify the extra nesting

Comment (sebthom) [reply README.md]: Am I discussing with claude or is there still a human involved? option 2 sounds good.

Comment (1cu) [reply README.md]: Hard to tell these days, right?
Let’s just say Claude was a lot more interesting to bounce ideas off than Codex, Cursor, or _worst of all_ CodeRabbit.

Option 2 was your suggestion, by the way. Claude/Codex/Cursor just evaluated it and cleaned it up.
I like it a lot better than before.

(Apple Intelligence threw in a few spelling suggestions.)

File: README.md [hunk 2]
```diff
@@ -377,6 +377,13 @@ category: # e.g. "Elektronik > Notebooks"
 
 price: # without decimals, e.g. 75
 price_type: # one of: FIXED, NEGOTIABLE, GIVE_AWAY (default: NEGOTIABLE)
+auto_price_reduction:
+  enabled: # true or false to enable automatic price reduction on reposts (default: false)
+  strategy: # "PERCENTAGE" or "FIXED" (required when enabled is true)
+  amount: # reduction amount; interpreted as percent for PERCENTAGE or currency units for FIXED
+  min_price: # required when enabled is true; minimum price floor (use 0 for no lower bound)
+  delay_reposts: # number of reposts to wait before first reduction (default: 0)
+  delay_days: # number of days to wait after publication before reductions (default: 0)
```
Comment (sebthom) [inline README.md:394]: how about instead of `delay_reposts` and `delay_days` we use something like:

```yaml
# First reduction gates
reposts_before_first_reduction: 2   # default: 0
days_before_first_reduction: 1      # default: 0

# Subsequent reduction pacing
reposts_between_reductions: 1       # default: 1 (reduce on every repost)
days_between_reductions: 0          # default: 0 (no delay)
```

WDYT?

Comment (1cu) [reply README.md:394]: Looking through the code we would need to add yet another field to calculate the  days between subsequent reductions. 

This feature is getting complex and I don't want to build all possible cases - not knowing what will be used at all.

File: src/kleinanzeigen_bot/model/ad_model.py [hunk 3]
```diff
@@ -106,21 +122,40 @@ def _validate_description_length(cls, v:str) -> str:
...
+        # Validate auto_price_reduction configuration
+        if auto_price_reduction and isinstance(auto_price_reduction, dict):
+            if auto_price_reduction.get("enabled"):
+                if price is None:
+                    raise ValueError("price must be specified when auto_price_reduction is enabled")
+                min_price = auto_price_reduction.get("min_price")
+                if min_price is not None and price is not None and min_price > price:
+                    raise ValueError("min_price must not exceed price")
+
         return values
```
[the remark this answers is not in the document]
Comment (1cu) [reply src/kleinanzeigen_bot/model/ad_model.py:179]: out of scope for this PR

**Review by sebthom — APPROVED** [review]
